code-challenges.js

```diff
@@ -3,6 +3,50 @@
 
 // --------------------1) Create a function that returns the first 10 numbers of the Fibonacci sequence in an array. Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+//fibonacci sequence - each subsequent number is the sum of the previous two
+// F(n) = F(n-1) + F(n-2)
+//
+
+// num1 = 1
+// num2 = 1
+// num3 = 2
+// num4 = 5
+
+//write a function that returns the number to 10 numbers in the array
+//the 10 numbers would go to index 9
+//the first number of fibonacci's sequence is actually 0 so need to start at index 1
+
+//first i wrote the logic for what is happening to the numbers in the fibonacci sequence, with regards to their index
+const fibonacci = (n) => {
+  if(n < 2){
+    return n
+  } else {
+    return fibonacci(n - 1) + fibonacci(n - 2)
+  }
+}
+
+//i was able to print the correct output for each index in the sequence, but I was stuck when it came to writing all of this in a single function
+console.log(fibonacci(2))
+console.log(fibonacci(6))
+console.log(fibonacci(0))
+
+//I thought it would make sense to wrap a for loop in a function, and then call the function that produces the fibonacci numbers
+i = 10
+const fibonacciArray = (i) => {
+  let newArray = []
+  for (let i=1; i<11; i++){
+    newArray.push(fibonacci(i))
+  }
+  return newArray
+}
+
+//this does print the array with the fibonacci sequence correctly, but I figured out the argument & variable i is pointless because you can set it to anything and it still returns indices 1-11 in accordance with the expected output.
+//you have to modify i within the for loop to return other parts of the fibonacci sequence if you want them
+//I feel like theres a better way to do this that combines it all into one function but I kept getting undefined
+console.log(fibonacciArray(i))
+
+
+
```

Comment:
Love the thoughtfulness of your answer. This problem is really challenging because it is "easy" for people to understand and really hard to describe in terms of code. I think you have all the pieces here, just need to put them together into one action.